vimeo-networking/src/main/java/com/vimeo/networking/model/VideoFile.java

```diff
@@ -24,21 +24,22 @@
 
 import com.google.gson.annotations.SerializedName;
 import com.vimeo.stag.UseStag;
+import com.vimeo.stag.UseStag.FieldOption;
 
+import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
 import java.io.Serializable;
 import java.util.Date;
 
 /**
- * Representation of a Video stream/playback file
+ * Base representation of a Video stream/playback file
  * <p>
  * Created by alfredhanssen on 4/25/15.
  */
 @SuppressWarnings("unused")
-// TODO: Remove the VideoFileDeserializer and use Stag instead, once API corrects log issue behind version 2/16/17 [KZ]
-// @UseStag(FieldOption.SERIALIZED_NAME)
-public class VideoFile implements Serializable {
+@UseStag(FieldOption.SERIALIZED_NAME)
+public abstract class VideoFile implements Serializable {
```

Comment:
Sounds like the choices are the client either needs to check an enum value then somehow know which fields are available based on the enum val or option 2; do and `instanceof` check  + a cast and then they can safely know which fields to use.  I prefer option 2 (keeping changes made in this PR).